hydromt/data_catalog/sources/geodataset.py

```diff
@@ -134,6 +134,10 @@ def to_file(
             handle_nodata=handle_nodata,
         )
         if ds is None:  # handle_nodata == ignore
+            exec_nodata_strat(
+                handle_nodata,
+                f"Reading file(s) for {self.name} returned no data.",
+            )
```

Comment:
Did you check the log? Feels like we might be warning twice? Already in read_data just above and now here again?